Code/GraphMol/Aromaticity.cpp

```diff
@@ -744,14 +853,38 @@ int aromaticityHelper(RWMol &mol, unsigned int minRingSize,
 }  // end of anonymous namespace
 
 int setAromaticity(RWMol &mol, AromaticityModel model, int (*func)(RWMol &)) {
+  // FIX: we will assume for now that if the input molecule came
+  // with aromaticity information it is correct and we will not
+  // touch it. Loop through the atoms and check if any atom has
+  // arom stuff set.  We may want check this more carefully later
+  // and start from scratch if necessary
+  ROMol::AtomIterator ai;
+  for (ai = mol.beginAtoms(); ai != mol.endAtoms(); ai++) {
```

Comment:
Can these be ranges?

Reply:
it's worth taking a look at enabling range-based iteration in some future feature addition, but I don't think it's currently doable